router/src/main.c

```diff
@@ -179,43 +180,40 @@ static void daemon_process(const char *config_path, const char *python_pkgdir, b
 
 
             //
-            // If config path is not represented by its full path, then
-            // save current path before changing to /
+            // If config path is not a fully qualified path, then construct the
+            // fully qualified path to the config file.  This needs to be done
+            // since the daemon will set "/" to its working directory.
             //
             char *config_path_full = NULL;
             if (strncmp("/", config_path, 1)) {
-                char *cur_path = NULL;
-                size_t path_size = 256;
-                int getcwd_error = 0;
-                cur_path = (char *) calloc(path_size, sizeof(char));
+                size_t path_size = PATH_MAX;
+                char *cur_path = (char *) calloc(path_size, sizeof(char));
                 errno = 0;
 
                 while (getcwd(cur_path, path_size) == NULL) {
                     free(cur_path);
-                    if ( errno != ERANGE ) {
-                        // If unable to get current directory
-                        getcwd_error = 1;
-                        break;
+                    if (errno != ERANGE) {
+                        // Hard failure - can't recover from this
```

Comment:
This change does not seem particularly related to this PR. Or is it? If not, can we make this change in a different commit ?

Reply:
It is related - the address sanitizer threw a fit on that section of the code - detected a possible buffer overflow.   That change is the fix.